executor/linux/outputs.go

```diff
@@ -0,0 +1,158 @@
+		return err
+	}
+
+	return nil
+}
+
+// destroy cleans up outputs container after execution.
+func (o *outputSvc) destroy(ctx context.Context, ctn *pipeline.Container) error {
+	// exit if outputs container has not been configured
+	if len(ctn.Image) == 0 {
+		return nil
+	}
+
+	// update engine logger with secret metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus#Entry.WithField
+	logger := o.client.Logger.WithField("outputs", ctn.Name)
+
+	logger.Debug("inspecting outputs container")
+	// inspect the runtime container
+	err := o.client.Runtime.InspectContainer(ctx, ctn)
+	if err != nil {
+		return err
+	}
+
+	logger.Debug("removing outputs container")
+	// remove the runtime container
+	err = o.client.Runtime.RemoveContainer(ctx, ctn)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// exec runs the outputs sidecar container for a pipeline.
+func (o *outputSvc) exec(ctx context.Context, _outputs *pipeline.Container) error {
+	// exit if outputs container has not been configured
+	if len(_outputs.Image) == 0 {
+		return nil
+	}
+
+	logrus.Debug("running outputs container")
+	// run the runtime container
+	err := o.client.Runtime.RunContainer(ctx, _outputs, o.client.pipeline)
+	if err != nil {
+		return err
+	}
+
+	logrus.Debug("inspecting outputs container")
+	// inspect the runtime container
+	err = o.client.Runtime.InspectContainer(ctx, _outputs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// poll tails the output for sidecar container.
+func (o *outputSvc) poll(ctx context.Context, ctn *pipeline.Container) (map[string]string, map[string]string, error) {
+	// exit if outputs container has not been configured
+	if len(ctn.Image) == 0 {
+		return nil, nil, nil
+	}
+
+	// update engine logger with outputs metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus#Entry.WithField
+	logger := o.client.Logger.WithField("outputs", ctn.Name)
+
+	logger.Debug("tailing container")
+
+	// grab outputs
+	outputBytes, err := o.client.Runtime.PollOutputsContainer(ctx, ctn, "/vela/outputs/.env")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// grab masked outputs
+	maskedBytes, err := o.client.Runtime.PollOutputsContainer(ctx, ctn, "/vela/outputs/masked.env")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return toMap(outputBytes), toMap(maskedBytes), nil
+}
+
+// toMap is a helper function that turns raw docker exec output bytes into a map
+// by splitting on carriage returns + newlines and once more on `=`.
+func toMap(input []byte) map[string]string {
+	// carriage returns are included in the split because the exec config `TTY` value is set to true
+	lines := strings.Split(string(input), "\r\n")
+
+	m := make(map[string]string)
+
+	for _, line := range lines {
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) == 2 {
+			s := parts[1]
+			if !strings.Contains(parts[1], "\\\\n") {
+				s = strings.Replace(parts[1], "\\n", "\\\n", -1)
+			}
+
+			m[parts[0]] = s
+		}
+	}
+
+	return m
+}
```

Comment:
i think it's ok to be less forgiving than something like https://github.com/hashicorp/go-envparse/blob/main/envparse_test.go, but i do wonder if we should account for a little more?

also, line 150, we should change to use `.ReplaceAll` instead